tests/testthat/test-learn.R

```diff
@@ -1,49 +1,87 @@
 # FIXME: #313: learn only from current source
 
-test_that("Learning from MSSQL works?", {
-
+test_that("Standard learning from MSSQL (schema 'dbo') works?", {
 
+  skip_if_src_not("mssql")
   # dm_learn_from_mssql() --------------------------------------------------
-  src_mssql <- skip_if_error(src_test("mssql"))
+  src_mssql <- my_test_src()
 
   # create an object on the MSSQL-DB that can be learned
   if (!any(src_tbls(src_mssql) %>%
     grepl("^tf_1_", .))) {
     copy_dm_to(src_mssql, dm_for_filter(), temporary = FALSE, table_names = unique_db_table_name)
   }
 
-  dm_for_filter_mssql_learned <- dm_from_src(src_mssql)
+  dm_for_filter_mssql_learned_all <- dm_from_src(src_mssql)
 
-  def_learned_renamed_reclassed <-
-    dm_rename_tbl(
-      dm_for_filter_mssql_learned,
-      structure(src_tbls(dm_for_filter_mssql_learned), names = src_tbls(dm_for_filter()))
-    ) %>%
-    dm_get_def() %>%
-    select(-data)
-
-  def_original <-
-    dm_get_def(dm_for_filter()) %>%
-    select(-data)
-
-  expect_identical(
-    def_learned_renamed_reclassed,
-    def_original
+  # in case there happen to be other tables in schema "dbo"
+  dm_for_filter_mssql_learned <- dm_select_tbl(
+    dm_for_filter_mssql_learned_all,
+    which(grepl("tf_[1-6]_[0-9]{4}_[0-9_]{5}_[0-9]", names(dm_for_filter_mssql_learned_all)))
+  ) %>% dm_select_tbl(
+    tf_1 = starts_with("tf_1"), tf_2 = starts_with("tf_2"), tf_3 = starts_with("tf_3"),
+    tf_4 = starts_with("tf_4"), tf_5 = starts_with("tf_5"), tf_6 = starts_with("tf_6"))
+
+  expect_equivalent_dm(
+    dm_for_filter_mssql_learned,
+    dm_for_filter()
+  )
+
+  walk(
```

Comment:
This should be code in `on.exit()`, so that it is run also when the test aborts.

Reply:
Good catch

Reply:
FYI: I used `withr::defer()` in d07251a and the equivalent commit for the other test because I found this in the web: https://testthat.r-lib.org/articles/test-fixtures.html